src/jsonata/parser.py

```diff
@@ -1180,6 +1180,7 @@ def lambda1(terms):
                 result.position = expr.position
                 result.lhs = self.process_ast(expr.lhs)
                 result.rhs = self.process_ast(expr.rhs)
+                result.keep_array = result.lhs.keep_array or result.rhs.keep_array
```

Comment:
Perhaps this is the fix that is needed in jsonata.js?  If so, I'll wait until the fix is made there first.